src/components/EmojiList/EmojiList.jsx

```diff
@@ -0,0 +1,127 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import styles from './EmojiList.scss';
+import classnames from 'classnames/bind';
+
+const cx = classnames.bind( styles );
+
+const getBulletStyle = ( bullet, id, i ) => {
+	if ( ! bullet ) {
+		return null;
+	}
+
+	return `.${id} li:nth-of-type(${i + 1})::before { content: '${bullet}'; }`;
+};
+
+const ListStyles = ( { bullets, id } ) => (
+	<style
+		dangerouslySetInnerHTML={{
+			__html: bullets
+				.map( ( bullet, index ) => getBulletStyle( bullet, id, index ) )
+				.join( '' ),
+		}}
+	/>
+);
+
+ListStyles.propTypes = {
+	bullets: PropTypes.array.isRequired,
+	id: PropTypes.string.isRequired,
+};
+
+const ListItems = ( {
+	id,
+	innerHtml,
+	tagName,
+} ) => (
+	React.createElement(
+		tagName,
+		{
+			className: `${styles.container} ${id}`,
+			dangerouslySetInnerHTML: { __html: innerHtml },
+		}
+	)
+);
+
+ListItems.propTypes = {
+	id: PropTypes.string.isRequired,
+	innerHtml: PropTypes.string.isRequired,
+	tagName: PropTypes.string.isRequired,
+};
+
+// If the emojis array contains emojis, strip those emojis from the supplied HTML
+const stripHtmlEmojis = ( html, emojis ) => {
+	if ( ! emojis.length ) {
+		return html;
+	}
+
+	const pattern = new RegExp( `[${emojis.join( '' )}]`, 'g' );
+	return html.replace( pattern, '' );
+};
+
+/**
+ * Displays a WP content block.
+ */
+const EmojiList = ( {
+	bullets,
+	id,
+	innerHtml,
+	tagName,
+} ) => (
+	<Fragment>
+		<ListStyles bullets={bullets} id={id} />
+		<ListItems
+			id={id}
+			innerHtml={stripHtmlEmojis( innerHtml, bullets )}
+			tagName={tagName}
+		/>
+	</Fragment>
+);
+
+EmojiList.propTypes = {
+	bullets: PropTypes.arrayOf( PropTypes.string ).isRequired,
+	id: PropTypes.string.isRequired,
+	innerHtml: PropTypes.string.isRequired,
+	tagName: PropTypes.oneOf( [ 'ol', 'ul' ] ).isRequired,
+};
+
+EmojiList.defaultProps = {
+	bullets: [],
+	tagName: 'ul',
+};
+
+/**
+ * An emoji list of arbitrary content.
+ */
+export const StructuredEmojiList = ( {
+	items,
+	id,
+} ) => {
+	const listStyle = items
+		.map( ( item, i ) => getBulletStyle( item.bullet, id, i ) )
+		.join( '' );
+
+	return (
+		<ul className={cx( 'container', id )}>
+			<style dangerouslySetInnerHTML={{ __html: listStyle }}/>
+			{items.map( ( { item }, i ) => (
+				<li key={i} className={cx( id )}>
```

Comment:
It's a little weird to see the `id` prop being assigned as a class name. I have an idea, though—instead of using arbitrary IDs, why not use the emoji themselves as class names? E.g., for an EmojiList with three bullets of 😊, 😜, 😻, create a class name of:

```css
.😊-😜-😻 li:nth-of-type(...) {
/* stuff */
}
```

Emoji are valid CSS identifiers and outside of [some contankerous advice about code _sharing_](https://adrianroselli.com/2017/10/avoid-emoji-as-class-names.html) it seems to be supported and accepted.

Reply:
also confused about why you are binding the classname to the `ul` and the `li`

Reply:
Wow that is a _very_ fun idea. I like that. Uhm.....not gonna lie, brought this styling over. Happy to take another look and simplify a bit.

Reply:
Ok - added the emoji class to the `ul` and removed the original id class from the `li` (it wasn't doing anything there)